tests/System.Binary.Base64.Tests/BasicUnitTests.cs

```diff
@@ -186,6 +304,256 @@ public void DecodingOutputTooSmall()
         }
 
         [Fact]
+        public void SampleUsage1()
+        {
+            Span<byte> source = new byte[1000];
+            InitalizeDecodableBytes(source);
+
+            Span<byte> decodedBytes = new byte[Base64Encoder.ComputeDecodedLength(source)];
+
+            Assert.True(Base64Encoder.TryDecode(source, decodedBytes, out int consumed, out int written));
+            Assert.Equal(source.Length, consumed);
+            Assert.Equal(decodedBytes.Length, written);
+        }
+
+        [Fact]
+        public void SampleUsage2()
+        {
+            var rnd = new Random(42);
+            Span<byte> source = new byte[1000];
+            InitalizeDecodableBytes(source);
+
+            int numBytes = rnd.Next(100, 251);
+            Span<byte> decodedBytes = new byte[numBytes];
+
+            int bytesConsumed = 0;
+            int bytesWritten = 0;
+
+            bool success = false;
+            do
+            {
+                success = Base64Encoder.TryDecode(source.Slice(bytesConsumed), decodedBytes, out int consumed, out int written);
+                bytesConsumed += consumed;
+                bytesWritten += written;
+                numBytes = rnd.Next(100, 251);
+                decodedBytes = new byte[numBytes];
+            } while (!success);
+
+            Assert.Equal(source.Length, bytesConsumed);
+            Assert.Equal(Base64Encoder.ComputeDecodedLength(source), bytesWritten);
+        }
+
+        [Fact]
+        public void SampleUsage3()
```

Comment:
I find it hard to follow the tests. There are several of them and I am not sure what they test. I think it would be better to have:
1. One test that tests that a single buffer is encoded properly 
2. One test that tests that a single buffer is decoded properly
3. One tests for various error conditions (if any)
4. One tests that implements the piping loop with InlineData for various combinations of buffers.